src/test/java/io/wgj/ctci/ArrayListTest.java

```diff
@@ -0,0 +1,136 @@
+package io.wgj.ctci;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.Iterator;
+
+import static org.junit.Assert.*;
+
+public class ArrayListTest {
+  @Rule
+  public final ExpectedException exception = ExpectedException.none();
+
+  @Test
+  public void arrayList_new() {
+    ArrayList<Integer> al = new ArrayList<>();
+  }
+
+  @Test
+  public void arrayList_add() {
+    ArrayList<Integer> al = new ArrayList<>();
+    al.add(1);
+  }
+
+  @Test
+  public void arrayList_size() {
+    ArrayList<Integer> al = new ArrayList<>();
+    assertEquals(0, al.size());
+    al.add(1);
+    assertEquals(1, al.size());
+  }
+
+  @Test
+  public void arrayList_grow() {
+    ArrayList<Integer> al = new ArrayList<>();
+    assertEquals(0, al.size());
+    Integer[] ints = {1, 2, 3, 4, 5};
+    for (Integer i : ints) {
+      al.add(i);
+    }
+    assertEquals(5, al.size());
+  }
+
+  @Test
+  public void arrayList_get() {
+    ArrayList<Integer> al = new ArrayList<>();
+    al.add(1);
+    assertEquals((Integer) 1, al.get(0));
+  }
+
+  @Test
+  public void arrayList_rangeCheck() {
+    ArrayList<Integer> al = new ArrayList<>();
+    exception.expect(IndexOutOfBoundsException.class);
+    al.get(0);
+  }
+
+  @Test
+  public void arrayList_remove_front() {
+    ArrayList<Integer> al = new ArrayList<>();
+    Integer[] add_ints = {1, 2, 3, 4, 5};
+    for (Integer i : add_ints) {
+      al.add(i);
+    }
+    assertEquals(add_ints[0], al.remove(0));
+
+    Integer[] get_ints = {2, 3, 4, 5};
+    for (int i = 0; i <= get_ints.length - 1; i++) {
```

Comment:
I think in this comparison could just be i < get_ints.length?

Also consider making a helper method that takes an ArrayList and an array, ensuring the contents are equivalent.